pkg/model/pki.go

```diff
@@ -142,12 +142,12 @@ func (b *PKIModelBuilder) Build(c *fi.ModelBuilderContext) error {
 			Format:    format,
 		})
 
-		// @check if calico is enabled as the CNI provider
-		if b.KopsModelContext.Cluster.Spec.Networking.Calico != nil {
+		// @check if calico or Cilium is enabled as the CNI provider
+		if b.KopsModelContext.Cluster.Spec.Networking.Calico != nil || b.KopsModelContext.Cluster.Spec.Networking.Cilium != nil {
 			c.AddTask(&fitasks.Keypair{
-				Name:      fi.String("calico-client"),
+				Name:      fi.String("etcd-cni-client"),
```

Comment:
So this one is tricky.  We'll generate a new (client) keypair here.  Specifically for calico that means that the clients will change keys.

However, because it isn't the key itself that matters, but rather the CA, I think this new key will be fine and not interrupt anything.